pyomo/contrib/sensitivity_toolbox/tests/test_sens_unit.py

```diff
@@ -0,0 +1,822 @@
+        ptb_list = [model.perturbed_eta]
+
+        sens = SensitivityInterface(model, clone_model=False)
+        sens.setup_sensitivity(param_list)
+        sens.perturb_parameters(ptb_list)
+        instance = sens.model_instance
+        block = sens.block
+
+        param_var_map = ComponentMap((param, var)
+                for var, param, _, _ in sens.block._sens_data_list)
+        param_con_map = ComponentMap((param, block.paramConst[i+1])
+                for i, (_, param, _, _) in
+                enumerate(sens.block._sens_data_list))
+        for param, ptb in zip(param_list, ptb_list):
+            for idx in param:
+                obj = param[idx]
+                ptb_data = ptb[idx]
+                var = param_var_map[obj]
+                con = param_con_map[obj]
+                self.assertEqual(instance.sens_state_value_1[var],
+                        ptb_data.value)
+                self.assertEqual(instance.DeltaP[con], -delta)
+
+    def test_perturb_indexed_parameters_with_scalar(self):
+        model = make_indexed_model()
+        param_list = [model.eta]
+        ptb_list = [10.0]
+
+        sens = SensitivityInterface(model, clone_model=False)
+        sens.setup_sensitivity(param_list)
+        sens.perturb_parameters(ptb_list)
+        instance = sens.model_instance
+        block = sens.block
+
+        param_var_map = ComponentMap((param, var)
+                for var, param, _, _ in sens.block._sens_data_list)
+        param_con_map = ComponentMap((param, block.paramConst[i+1])
+                for i, (_, param, _, _) in
+                enumerate(sens.block._sens_data_list))
+        for param, ptb in zip(param_list, ptb_list):
+            for idx in param:
+                obj = param[idx]
+                var = param_var_map[obj]
+                con = param_con_map[obj]
+                self.assertEqual(instance.sens_state_value_1[var], ptb)
+                self.assertEqual(instance.DeltaP[con], obj.value-ptb)
+
+    def test_get_dsdp1(self):
+        '''
+        It tests the function get_dsdp with a simple nonlinear programming example.
+        
+        min f: p1*x1+ p2*(x2^2) + p1*p2 
+         s.t c1: x1 = p1
+             c2: x2 = p2
+             c3: 10 <= p1 <= 10
+             c4: 5 <= p2 <= 5  
+        '''
+        variable_name = ['p1', 'p2']
+
+        m= ConcreteModel()
+        m.x1 = Var(initialize = 0)
+        m.x2 = Var(initialize = 0)
+        m.p1 = Var(initialize = 0)
+        m.p2 = Var(initialize = 0)
+        m.obj = Objective(expr = m.x1*m.p1+m.x2*m.x2*m.p2 + m.p1*m.p2 , sense=minimize)
+        m.c1 = Constraint(expr = m.x1 == m.p1)
+        m.c2 = Constraint(expr = m.x2 == m.p2)
+        theta= {'p1': 10.0, 'p2': 5.0}
+        for v in variable_name:
+            getattr(m, v).setlb(theta[v])
+            getattr(m, v).setub(theta[v])
+        dsdp_dic, col = get_dsdp(m, variable_name, theta)
+        np.testing.assert_almost_equal(dsdp_dic['d(x1)/d(p1)'], 1.0)
```

Comment:
This test appears to be failing

Reply:
1.0 is a correct value and 1.0 is ok in my local machine. I remembered it was ok when I first made this test. I don't know why it changed to an error. I will check it to fix the error.

Reply:
Is this the off-by-one error in k_aug that Alex told me something about? I think I recently updated and re-compiled k_aug locally but I can check again. This might explain the GJH errors above as well.
If it is we'll need to get John Eslick to update the Idaes-extensions k_aug build.

Reply:
That was an error for `A_print.txt`. John Eslick updated it, https://github.com/IDAES/idaes-pse/pull/266. Pyomo needs to update it as well?